Basic auth support

johanlundberg opened the pull request.

johanlundberg committed:
Added support for basic auth header

johanlundberg committed:
Basic auth for tests and paver

johanlundberg committed:
Seems like an error automatically rolls back the transaction now

jakewins commented:
Looks like some of the python versions are not happy with `self._COMMON_HEADERS.update({"Authorization": "Basic {}".format(auth)})` - but other than that, LGTM! Happy to merge this once the build failures are resolved.

johanlundberg committed:
Python 2.6 and 3 support

base64 encode basic auth for both python 2 and 3
Sometimes the server isnt ready, added retry
Added sudo: false in .travis.yml to use updated travis infrastructure

johanlundberg commented:
Oops :)

My PR should now take all versions of Python in to account.
